### DISCUSSION TITLE
Is there any due date for 2.0.0-M1 ?

### DISCUSSION THREAD

**Discussion by Alessandrovito** [spring-projects/spring-retry#306] [Q&A] [answered]
Since we are affected by a bugs present in version 1.3.3 version ( issues/184) already fixed and merged to the master branch , Do you have any due date for next release 2.0.0-M1 ? Thanks

**Comment by snicoll** [comment, marked as answer]
I've scheduled M1 for September 16th.

**Reply by Alessandrovito** [reply]
Cool ! Looking forward to use your new version when it will come out ! Thanks for the heads up.

**Reply by Alessandrovito** [reply]
Hi, the 2.0.0-M1  is out marked as  "Pre-release". Will this version be available in maven central  (https://mvnrepository.com/artifact/org.springframework.retry/spring-retry) ? In case, how long time it will take to be present in maven central ? Thanks

**Reply by snicoll** [reply]
It's a milestone not a GA release. It won't be available on maven central but on our milestone repo.

**Reply by Alessandrovito** [reply]
If I want to use your milestone version in my pom file, what is your recommendation ?

**Reply by snicoll** [reply]
I wouldn't recommend using the milestone in production. You can give it a try adding https://repo.spring.io/milestone to your build.

**Reply by garyrussell** [reply]
>Since we are affected by a bugs present in version 1.3.3 version ( issues/184) already fixed...

#184 is a new feature, not a bug.

1.3.x is still java6 based so we can't use `java.util.Supplier` there (it was introduced in Java 8).

 Using a supplier is the mechanism by which we can evaluate the expression at runtime.

We can't force all 1.3.x users to use a different version of java.

**Reply by Alessandrovito** [reply]
@snicoll When the  2.0.0-M1 will be promoted to become an official release presents in Maven Central ?

**Reply by garyrussell** [reply]
It won't; it's a milestone towards the 2.0 GA in November.

**Reply by Alessandrovito** [reply]
Ok, then when will come out a new official version present in Maven Central that will include the https://github.com/spring-projects/spring-retry/issues/184 ?

**Reply by snicoll** [reply]
Late November at the latest.

**Reply by Alessandrovito** [reply]
Thanks
